src/main/java/ru/l3r8y/prefixed/PrefixedCop.java

```diff
@@ -23,21 +23,130 @@
  */
 package ru.l3r8y.prefixed;
 
+import io.github.classgraph.AnnotationInfo;
+import io.github.classgraph.AnnotationParameterValue;
+import io.github.classgraph.ClassGraph;
+import io.github.classgraph.ClassInfo;
+import io.github.classgraph.ClassInfoList;
+import io.github.classgraph.ScanResult;
 import org.apache.maven.plugin.AbstractMojo;
 import org.apache.maven.plugin.MojoExecutionException;
-import org.apache.maven.plugin.MojoFailureException;
 import org.apache.maven.plugins.annotations.LifecyclePhase;
 import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+import ru.l3r8y.prefixed.annotation.Prefixed;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
 
 /**
  * Prefixed cop.
+ * This class designed with focus on performance, this is
+ * why code may look ugly to you.
+ *
  * @since 0.0.0
  */
 @Mojo(name = "prefixed", defaultPhase = LifecyclePhase.VERIFY)
 public final class PrefixedCop extends AbstractMojo {
 
+    /**
+     * Thread pool to scan project.
+     */
+    private static final ExecutorService EXECUTORS = Executors.newVirtualThreadPerTaskExecutor();
+
+    /**
+     * The project.
+     */
+    @Parameter(defaultValue = "${project}")
+    private MavenProject project;
+
+    /**
+     * The base package.
+     */
+    @Parameter(property = "basePackage", defaultValue = "${project.groupId}")
+    private String basePackage;
+
+    /**
+     * Fail on error flag.
+     */
+    @SuppressWarnings("PMD.ImmutableField")
+    @Parameter(defaultValue = "true")
+    private boolean failOnError = true;
+
     @Override
-    public void execute() throws MojoExecutionException, MojoFailureException {
-        this.getLog().info("Boom!");
+    public void execute() throws MojoExecutionException {
+        final String outputDir = this.project.getBuild().getOutputDirectory();
+        final File output = new File(outputDir);
+        if (!output.exists()) {
+            this.getLog().info(String.format("Output directory does not exist: '%s'", outputDir));
+            return;
+        }
+        this.getLog().info(String.format("Scanning compiled classes in: '%s'", outputDir));
+        final ClassGraph graph = new ClassGraph()
+            .overrideClasspath(outputDir)
+            .acceptPackages(this.basePackage)
+            .enableClassInfo()
+            .enableAnnotationInfo();
+        try (final ScanResult scanResult = graph.scan(PrefixedCop.EXECUTORS, Runtime.getRuntime().availableProcessors())) {
+            final List<ClassInfo> prefixedInterfaces = scanResult
+                .getClassesWithAnnotation(Prefixed.class)
+                .stream()
+                .filter(ClassInfo::isInterface)
+                .toList();
+            this.getLog().info("Total interfaces found: " + prefixedInterfaces.size());
```

Comment:
@l3r8yJ let's use `String.format()` instead of concatenation. Also, read this: https://www.yegor256.com/2014/06/19/avoid-string-concatenation.html